packages/@lwc/engine-core/src/framework/rendering.ts

```diff
@@ -282,10 +286,20 @@ function mountCustomElement(
     insertNode(elm, parent, anchor, renderer);
 
     if (vm) {
-        if (process.env.NODE_ENV !== 'production') {
-            assert.isTrue(vm.state === VMState.created, `${vm} cannot be recycled.`);
+        if (process.env.IS_BROWSER) {
+            if (!features.ENABLE_NATIVE_CUSTOM_ELEMENT_LIFECYCLE) {
+                if (process.env.NODE_ENV !== 'production') {
```

Comment:
Forgot to remove the other assertion at L278-280?

Reply:
Yes. 🤦